WordPress/gutenberg
# Font Appearance Control: Refactor font appearance fallbacks

## What?

Refactors the logic around the font weight and style fallbacks applied when a new font family is selected. Follow-up to https://github.com/WordPress/gutenberg/pull/61915.

## Why?

Addresses this comment: https://github.com/WordPress/gutenberg/pull/61915#discussion_r1666394116.

## How?

Adds a new `findNearestStyleAndWeight` function that finds the nearest available weight and style to apply when the font family is changed, regardless of whether the font family is selected using global styles or dynamically via a new block. Also, as the logic now sits in a function, it is easier to add automated tests.

## Testing Instructions




Ensure there are a variety of font faces installed in multiple font families, including different font weights, styles and at least one system font.

1. Open Global Styles > Typography
2. Change the active font by using the Font dropdown menu to a font that has different weights/styles
3. Ensure that the nearest available appearance options are selected automatically, e.g. the closest available font weight and the same font style where available

Try running the following in the console:

```
var block = wp.blocks.createBlock('core/paragraph', {
  content: 'Hello',
  style: { typography: { fontWeight: '300' } } }
);
wp.data.dispatch('core/block-editor').insertBlock( block )
```

The nearest available font weight and style should be automatically applied.

## Screenshots or screencast
